gridappsd/field_interface/context.py

```diff
@@ -1,36 +1,53 @@
-from gridappsd import GridAPPSD
+from gridappsd.field_interface.interfaces import FieldMessageBus
+import gridappsd.topics as t
 
-request_field_queue_prefix = 'goss.gridappsd.process.request.field'
-request_field_context_queue = request_field_queue_prefix + '.context'
 
 
-class ContextManager:
-
+class LocalContext:
+    
     @classmethod
-    def get_context_by_feeder(cls, feeder_mrid, area_id=None):
-        gridappsd_obj = GridAPPSD()
-
-        request = {'modelId': feeder_mrid,
+    def get_context_by_feeder(cls, downstream_message_bus: FieldMessageBus, feeder_mrid, area_id=None):
+        
+        request = {'request_type' : 'get_context',
+                    'modelId': feeder_mrid,
                    'areaId': area_id}
-
-        response = gridappsd_obj.get_response(request_field_context_queue, request)
+        response = downstream_message_bus.get_response(t.context_request_queue(downstream_message_bus.id), request, timeout=10)
         return response
 
     @classmethod
-    def get_context_by_message_bus(cls, downstream_message_bus_id):
+    def get_context_by_message_bus(cls, downstream_message_bus: FieldMessageBus):
         """
         return agents/devices based on upstream and/or downstream message bus as input
```

Comment:
the comment indicates that you can use the upstream/or downstream bus, however the argument seems to suggest only downstream is available.

Reply:
Comment changed.